core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3870,7 +3870,7 @@ private static AtomicLong getAtomicLong(String key) {
   }
 
   /** Support the ARRAY_COMPACT function. */
-  public static List compact(List list) {
+  public static List arrayCompact(List list) {
```

Comment:
why are we going to rename public methods?
that could impact downstream dependencies

Reply:
@snuyanzin this method merged in the same version, just a few days, do not have released

Reply:
ok then the naming became more confusing: `arrayCompact` says about array however returns a `List`

Reply:
@snuyanzin just for unify with others, i use removeNulls before, but change it to arrayCompact when reviewed.

Reply:
I didn't get it

I see a review comment here https://github.com/apache/calcite/pull/3233#discussion_r1210553037
to use `compact` name and from my point of view this is fine since with such name there is no confusion about `List` and `array`
Where in review `arrayCompact` has appeared?

Reply:
@snuyanzin 
"this could also be named compact so it's easier to find, removeNulls is also ok", this is in the comments you refer.
originally i use removeNulls then changed to compact, and i changed to arrayCompact just because the spark function name is arrayCompact, which others do it too. for search easy.
https://github.com/apache/calcite/pull/3212#discussion_r1213835525

Reply:
my concern is not about search, moreover i don't think search is an issue since you have a comment with sql version
my concern is about name confusion which I raised in this thread
> ok then the naming became more confusing: `arrayCompact` says about `array` however returns a `List`

Reply:
@snuyanzin i know you meaning. List is just array type's runtime implementation. from my side, the name is ok. if you think it is not good, i can change all of them to revert.

Reply:
hi @JiajunBernoulli what is your opinion, should it revert? if do don't oppose it, i will revert according to @snuyanzin

Reply:
```
  ARRAY_MAX(SqlFunctions.class, "arrayMax", List.class),
  ARRAY_MIN(SqlFunctions.class, "arrayMin", List.class),
```
should be save.

Other function name can be revert.

Reply:
@JiajunBernoulli has reverted, thanks for your suggestion @JiajunBernoulli @snuyanzin